### PR TITLE
Add session cleanup changes to CHANGELOG.md

### PR DESCRIPTION
**Pull request by vancluever**
This adds the changelog entries associated with the following PRs:

#1340 
#1354
#1400
#1410

### CODE REVIEW

**Review by jefferai — DISMISSED** [review]
Looks good. One comment: in the first bullet point, the last sentence could be seen to mean that workers won't close connections _until_ the status update comes after connectivity is restored. Might be better to instead note that it will be cleaned up at that point, but will have already been closed.

**Comment by vancluever** [comment]
I think I'm just going to strike that part as I think the extra context can be confusing, I can't seem to find a good way to explain it more without writing a book. 😜 I think just saying that the controller is now marking the connections closed like the worker does should be good enough.

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,17 @@ Canonical reference for changes, improvements, and bugfixes for Boundary.
   it significantly easier to write filters to create [managed
   groups](https://www.boundaryproject.io/docs/concepts/filtering/oidc-managed-groups).
   ([PR](https://github.com/hashicorp/boundary/pull/1419))
+* Controllers will now mark connections as closed in the database if the worker
+  has not reported its status; this can be seen as the controller counterpart to
+  the worker-side session cleanup functionality released in 0.4.0. As with the
+  worker, this timeout for this behavior is 15s.
```
Comment (jimlambrt) [inline CHANGELOG.md:36]:
```suggestion
  worker, the timeout for this behavior is 15s.
```

**Review by jimlambrt — CHANGES_REQUESTED** [review]
one small suggestion

**Review by jimlambrt — APPROVED** [review]

**Pull request merged**